### PR TITLE
Add SNAT enabled bool for DHCP create function

### PR DESCRIPTION
**Pull request by rmoralesjr**
Add SNAP enabled bool for DHCP create function

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

=== RUN   TestAccIBMPIDhcpSNAT
--- PASS: TestAccIBMPIDhcpSNAT (374.02s)
PASS
```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_dhcp.go [hunk 1]
```diff
@@ -65,6 +65,13 @@ func ResourceIBMPIDhcp() *schema.Resource {
 				Description: "Optional name of DHCP Service (will be prefixed by DHCP identifier)",
 				ForceNew:    true,
 			},
+			Arg_DhcpSnatEnabled: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Indicates if SNAT will be enabled for the DHCP service",
+				ForceNew:    true,
+			},
```
Comment (hkantare) [inline ibm/service/power/resource_ibm_pi_dhcp.go:74]: Can we set d.Set in read method

Comment (rmoralesjr) [reply ibm/service/power/resource_ibm_pi_dhcp.go:74]: At this time the service broker does not provide a SNAT enabled value on the DHCP server read endpoint. This was done by design per the service broker developer.

File: ibm/service/power/resource_ibm_pi_dhcp_test.go [hunk 2]
```diff
@@ -133,3 +133,30 @@ func testAccCheckIBMPIDhcpWithCidrNameConfig(name string) string {
...
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIDhcpConfigWithSNATDisabled() string {
+	return fmt.Sprintf(`
+	resource "ibm_pi_dhcp" "dhcp_service" {
+		pi_cloud_instance_id = "%s"
+		pi_dhcp_snat_enabled = "false"
```
[the remark this answers is not in the document]
Comment (rmoralesjr) [reply ibm/service/power/resource_ibm_pi_dhcp_test.go]: Ok I updated the test to the suggested change. Thanks.

--- PASS: TestAccIBMPIDhcpSNAT (471.65s)
PASS

**Review by yussufsh — APPROVED** [review]
lgtm

**Comment by rmoralesjr** [comment]
@hkantare Any other concerns with this PR or can we merge it? Thank you.

**Pull request merged**